Source/Diagnostics/BoostedFrameDiagnostic.cpp

```diff
@@ -446,6 +444,36 @@ namespace
 }
 #endif
 
+namespace
+{
+    void
+    CopySlice(MultiFab& tmp, MultiFab& buf, int k_lab, 
+              std::vector<int> map_actual_fields_to_dump)
+    {
+        const int ncomp_to_dump = map_actual_fields_to_dump.size();
+        // Copy data from MultiFab tmp to MultiFab data_buffer[i].
+        for (MFIter mfi(tmp, TilingIfNotGPU()); mfi.isValid(); ++mfi) {
+            Array4<      Real> tmp_arr = tmp[mfi].array();
+            Array4<      Real> buf_arr = buf[mfi].array();
+            // For 3D runs, tmp is a 2D (x,y) multifab, that contains only 
+            // slice to write to file.
+            const Box& bx  = mfi.tilebox();
+            
+            ParallelFor(bx, ncomp_to_dump,
```

Comment:
Did you verify that this works for both 2D and 3D?

Reply:
Yep, it does.